src-tauri/src/periodic/config.rs

```diff
@@ -0,0 +1,185 @@
+use std::{collections::HashSet, str::FromStr, time::Duration};
+use tauri::{AppHandle, Manager, State, Url};
+use tokio::time::sleep;
+
+use crate::{
+    appstate::AppState,
+    commands::{device_config_to_location, update_instance},
+    database::{
+        models::{Id, NoId},
+        DbPool, Instance, Location, WireguardKeys,
+    },
+    error::Error,
+    events::{CONFIG_CHANGED, INSTANCE_UPDATE},
+    proto::{DeviceConfigResponse, InstanceInfoRequest, InstanceInfoResponse},
+};
+
+const INTERVAL_SECONDS: Duration = Duration::from_secs(30);
+const POLLING_ENDPOINT: &str = "/api/v1/poll";
+
+/// Periodically retrieves and updates configuration for all [`Instance`]s.
+/// Updates are only performed if no connections are established to the [`Instance`],
+/// otherwise event is emmited and UI message is displayed.
+pub async fn poll_config(handle: AppHandle) {
+    let state: State<AppState> = handle.state();
+    let pool = state.get_pool();
+    loop {
+        let Ok(instances) = Instance::all(&pool).await else {
+            error!("Failed to retireve instances, retrying in {}s", INTERVAL_SECONDS.as_secs());
+            sleep(INTERVAL_SECONDS).await;
+            continue;
+        };
+        debug!(
+            "Polling configuration updates for {} instances",
+            instances.len(),
+        );
```

Comment:
```suggestion
        if instances.is_empty() {
            debug!(
                "No instances found, skipping configuration polling, sleeping {}s",
                INTERVAL_SECONDS.as_secs()
            );
            sleep(INTERVAL_SECONDS).await;
            continue;
        }
        debug!(
            "Polling configuration updates for {} instances",
            instances.len(),
        );
```

maybe we should check if there are any instances at all? If someone uses the client purely for the tunnel functionality (as a "vanilla" wg client) the info log "Retrieved configuration for {} instances, sleeping {}s" would be still displayed every 30s

Reply:
I think it's fine, I'll take it over repeated sleep & continue.